packages/stake/src/stake.ts

```diff
@@ -36,3 +38,97 @@ export const modifyStakes = async (
 export const modifyStake = async (stake: Stake, options: Options) => {
   return modifyStakes([stake], options)
 }
+
+export const getUserJuiceDelta = async (
+  userAddress: string,
+  from?: string | number,
+  to?: string | number,
+  options?: Options,
+): Promise<BigNumber> => {
+  const contract = getJuiceStakingContract(options)
+  const deltaByToken: { [token: string]: BigNumber } = {}
+  const latestUnstakeByToken: { [token: string]: number } = {}
+  let delta = BigNumber.from(0)
+
+  const stakeFilter: ethers.EventFilter =
+    contract.filters.StakeAdded(userAddress)
+  const unStakeFilter: ethers.EventFilter =
+    contract.filters.StakeRemoved(userAddress)
+
+  const unStakes: ethers.Event[] | TypedEvent<any[]>[] =
+    await contract.queryFilter(unStakeFilter, from, to)
+
+  // First, filter every realized JUICE profit
+  unStakes.forEach((event) => {
+    if (event?.args && event?.blockNumber) {
+      const { args, blockNumber } = event
+      const { unstakedDiff, token } = args
+
+      if (deltaByToken[token]) {
+        deltaByToken[token] = deltaByToken[token].add(unstakedDiff)
+      } else {
+        deltaByToken[token] = unstakedDiff
+      }
+
+      if (
+        !latestUnstakeByToken[token] ||
+        latestUnstakeByToken[token] < blockNumber
+      ) {
+        latestUnstakeByToken[token] = blockNumber
+      }
+    }
+  })
+
+  // TODO: Currently the function might result in an edge case where it might be unfair for users that have staked right after 'from' block height against users that haven't. How to tackle this?
```

Comment:
Is this TODO still valid? I don't fully understand the unfairness :thinking:

Reply:
It is now, if someone has had a stake before the first unstake in the specified time frame. One way to remove this is to filter all JUICE deltas from the first unstake that has occurred inside the queried timeframe rather than a hard blockNumber, when we can encounter stakes that have occurred before the first unstake. Get it? :P

Reply:
Then the thing would be more fair, as calculation starts from the first unstake for all. It could be gamed though, but that isn't really a huge problem